lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriters.java

```diff
@@ -125,21 +126,21 @@ static PrimitiveValueWriter getValueWriter(ValuesWriter valuesWriter, Type type,
             return new TimeMicrosValueWriter(valuesWriter, parquetType);
         }
         if (type instanceof TimestampType) {
+            checkArgument(parquetTimeZone.isPresent(), "parquetTimeZone must be provided for timestamps");
             if (parquetType.getPrimitiveTypeName().equals(PrimitiveType.PrimitiveTypeName.INT96)) {
-                checkArgument(parquetTimeZone.isPresent(), "parquetTimeZone must be provided for INT96 timestamps");
                 return new Int96TimestampValueWriter(valuesWriter, type, parquetType, parquetTimeZone.get());
             }
             if (TIMESTAMP_MILLIS.equals(type)) {
                 verifyParquetType(type, parquetType, TimestampLogicalTypeAnnotation.class, isTimestamp(LogicalTypeAnnotation.TimeUnit.MILLIS));
-                return new TimestampMillisValueWriter(valuesWriter, type, parquetType);
+                return new TimestampMillisValueWriter(valuesWriter, type, parquetType, parquetTimeZone.get());
```

Comment:
We need to update `io.trino.parquet.writer.ParquetSchemaConverter#getPrimitiveType` to set isAdjustedToUTC to true in `LogicalTypeAnnotation.timestampType` if non-UTC timezone is configured.  ParquetSchemaConverter should always receive UTC from non-hive connectors.